### DISCUSSION TITLE
Ability to present roller shutter mode as Garage Door Opener to HomeKit instead of Window Covering

### DISCUSSION THREAD

**Discussion by DougMunford** [mongoose-os-apps/shelly-homekit#1474] [Ideas]
I have a shelly 2.5 in roller shutter mode controlling my garage door with firmware 2.9. Recently I added a few HomePods to my house and now realise that yelling "Hey Siri, open the garage door" near a window can activate a listening HomePod and result in an open garage. While I can set notifications on in the home app for any window/blind (which the roller shutter appears as) that only lets me know so I can close it again, it doesn't stop this sort of use. I could stop HomePods listening for "Hey Siri" and only respond to touch but that adds friction to other automation/requests that are not so security sensitive.

I note that HomeKit locks can require additional auth via touchID/faceID or PIN before they open and when HomePods are asked to control these devices they demur in favour of iPhones etc. Is this functionality something that is defined by an accessory and if so could it be added for particular use cases such as security shutters, garage doors or gate controls?

**Comment by rojer** [comment]
if it is, it's not described in the spec

**Comment by timoschilling** [comment]
HomeKit has this "Administrator Only Access", the specs saying the following:
> When this mode is enabled, the accessory only accepts administrator access.

Maybe it can be used in a garage door opener, to achieve the feature request. Will test it later today.

**Comment by DougMunford** [comment]
> if it is, it's not described in the spec

However the operation is described in the [HomePod user guide](https://support.apple.com/en-gb/guide/homepod/apd8085c1032/homepod)

_"Check the status of a security accessory: You can use HomePod to lock or check the status of door locks and other HomeKit security accessories. Say something like “Hey Siri, is the front door locked?” To unlock security accessories, you must use your iOS or iPadOS device because unlocking requires authentication using Face ID, Touch ID or a passcode."_

Is it implicit for locks and doors while the roller shutter mode is currently being defined as a window?

I did notice that locks have a special vendor defined [management control point](https://developer.apple.com/documentation/homekit/hmcharacteristictypelockmanagementcontrolpoint) characteristic but not sure exactly how that would be used.

**Comment by timoschilling** [comment]
> roller shutter mode controlling my garage door

Ah ok there is the point, I think. Steve would say "You holding it wrong!". Not testet with HomePod yet, but what I know from the iPhone is that a garage door opener is only controllable with an unlocked iPhone (Siri in lock mode don't work), but a window cover always works.

Is there a reason why you use the window cover mode and not the garage door opener mode?

**Comment by DougMunford** [comment]
My garage door is a large roller shutter type. I have wired the Shelly 2.5 in roller shutter mode so the relays are controlling the roller motor directly. There is no other control unit for the door, only the Shelly. Unless I missed something on setup in this firmware it appears roller shutter mode can only create windows, not doors?

**Comment by timoschilling** [comment]
Ok, seams legit, maybe we should a mode which acts as a WC on the shelly output side and as GDO on the HomeKit side.

What do you think @rojer?

**Comment by DougMunford** [comment]
That sounds sensible. It would remove the partial open/close functionality but I never use that anyway. How would that interact with the obstruction detection?

**Comment by DougMunford** [comment]
Just checking homebridge-Shelly for how they handle this on the 2.5 and they have made the presentation configurable. I presume these are the fixed HomeKit types and this firmware is using “WindowCovering”?

https://github.com/alexryd/homebridge-shelly#shelly-25-configurations

**Comment by DougMunford** [comment]
Just as a test I returned to stock firmware, set the 2.5 up in roller shutter mode again and added it to homebridge via the homebridge-Shelly plugin and override the type as “garageDoorOpener”. It appears in HomeKit as a door now and works fine. Siri requires device unlock before it will perform an instruction to open which is what we expected. I lose the positional control but that doesn’t matter. So, if there is a way to achieve this directly though the firmware it would be greatly appreciated.

**Comment by marcus-aa** [comment]
This would make a lot of sense, I see a lot of garage doors based on roller motors in some countries.

**Comment by PADE86** [comment]
Hello, are you still working on this topic? A Realisation would be great!
